src/app/modules/spcp/spcp.service.ts

```diff
@@ -77,4 +89,67 @@ export class SpcpService {
       return err(new CreateRedirectUrlError())
     }
   }
+
+  fetchLoginPage(
+    redirectUrl: string,
+  ): ResultAsync<string, FetchLoginPageError> {
+    return ResultAsync.fromPromise(
+      axios
+        .get<string>(redirectUrl, {
+          headers: {
+            Accept: LOGIN_PAGE_HEADERS,
+          },
+          timeout: LOGIN_PAGE_TIMEOUT,
+          // Throw error if not status 200.
+          validateStatus: (status) => status === StatusCodes.OK,
+        })
+        .then((response) => response.data),
+      (error) => {
+        logger.error({
+          message: 'Error while fetching SP/CP login page',
+          meta: {
+            action: 'fetchLoginPage',
+            redirectUrl,
+          },
+          error,
+        })
+        return new FetchLoginPageError()
+      },
+    )
+  }
+
+  validateLoginPage(
+    loginHtml: string,
+  ): Result<LoginPageValidationResult, LoginPageValidationError> {
+    // The successful login page should have the title 'SingPass Login'
+    // The error page should have the title 'SingPass - System Error Page'
+    const title = getSubstringBetween(loginHtml, '<title>', '</title>')
```

Comment:
I would like to propose removing Singpass e-service ID validation functionality. as you can see, the validation is tightly coupled to Singpass' UI, and they can break our app if they change the HTML title of their page.

Reply:
yeah I agree the implementation isn't ideal. on the other hand, I think the UX will be quite poor if we remove validation, because then admins who mistype their e-service IDs might publish forms where no one can log in. let's discuss this later?

I'm going to go ahead and merge this first, it's going into a feature branch and we can always roll it back before it goes into `develop`.